CONTRIBUTING.md

```diff
@@ -86,10 +86,36 @@ component of the Thanos distributed system. We recommend:
 `make help` will print most of available commands with relevant details.
 * Spin up a prebuilt dev environment using Gitpod.io
 [![Gitpod Ready-to-Code](https://img.shields.io/badge/Gitpod-ready--to--code-blue?logo=gitpod)](https://gitpod.io/#https://github.com/thanos-io/thanos)
+* In case you want to develop the project locally, install **Golang** in your machine. Here is a nice [gist](https://gist.github.com/nikhita/432436d570b89cab172dcf2894465753) for this purpose. -
+
+### Installing Project locally in your machine
+
+* Change your directory to the gopath for installing Thanos there -
+  * `cd $GOPATH/src/github.com/`
+* If you don’t have `$GOPATH` set, then add them -
+  * `export GOPATH=$HOME/go; export PATH=$PATH:$GOROOT/bin:$GOPATH/bin`
+  * Make sure you add the exports to the `/.bashrc` file so that those environment variables are persisted when you run them again.
+* If `github.com` folder is not present in the `src` path of `$GOPATH`, make one and clone Thanos inside it -
+  * For HTTPS - `git clone https://github.com/thanos-io/thanos.git`
+  * For SSH - `git clone git@github.com:thanos-io/thanos.git`
+* Once installed, you can run `make help` inside Thanos folder for getting a list of helper commands which are provided for making development easy for you :)
+
+### Building/Running/Developing
+
+* Run `make help` for getting a list of helper commands that will make your development life much more easy. It also provides auto **linting** and **formatter** for making sure the code quality meets the standards of contribution.
+
+* Usually, while sending in a PR  `make build`, `make format`, `make lint`, `make test`, `make docs`, `make check-docs` are the most used commands while developing Thanos.
+
+* When you run `make build` the code is compiled and a binary named `thanos` is created, which can be run to start `thanos` binary. To run the binary, use `./thanos`
+
+* In case you are working on a component of Thanos, you would love it if you don’t have to set up the yaml configuration for Prometheus and other components, before you start running the component. This is a repetitive task, and the Thanos Community has provided a shell script for automating the running the components -
+  * [quickstart.sh](https://github.com/thanos-io/thanos/blob/b08c0ea62abfe4dcf1400da0e37598f0cd8fa8cf/scripts/quickstart.sh) - run the all the components using `./quickstart`
```

Comment:
Using `quickstart.sh` also requires prometheus binary file. I think it would be nice to mention it.

Reply:
Or we should mention `make quickstart`, which takes care of installing Prometheus and other dependencies.

Reply:
Nice, didn't knew that we have a `make quickstart` command, adding it to the docs!